registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/main/java/io/mosip/registration/processor/quality/classifier/stage/QualityClassifierStage.java

```diff
@@ -1,492 +1,493 @@
+					.trimExceptionMessage(StatusUtil.UNKNOWN_EXCEPTION_OCCURED.getMessage() + ex.getMessage()));
+			registrationStatusDto.setSubStatusCode(StatusUtil.UNKNOWN_EXCEPTION_OCCURED.getCode());
+			registrationStatusDto.setLatestTransactionStatusCode(
+					registrationStatusMapperUtil.getStatusCode(RegistrationExceptionTypeCode.EXCEPTION));
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					regId,
+					RegistrationStatusCode.FAILED.toString() + ex.getMessage() + ExceptionUtils.getStackTrace(ex));
+			object.setInternalError(Boolean.TRUE);
+			description.setCode(PlatformErrorMessages.RPR_BDD_UNKNOWN_EXCEPTION.getCode());
+			description.setMessage(PlatformErrorMessages.RPR_BDD_UNKNOWN_EXCEPTION.getMessage());
+		} finally {
+			if(object.getInternalError()) {
+				updateErrorFlags(registrationStatusDto, object);
+			}
+			object.setRid(registrationStatusDto.getRegistrationId());
+			registrationStatusDto.setRegistrationStageName(getStageName());
+			registrationStatusDto
+					.setLatestTransactionTypeCode(RegistrationTransactionTypeCode.QUALITY_CLASSIFIER.toString());
+			String moduleId = isTransactionSuccessful ? PlatformSuccessMessages.RPR_QUALITY_CHECK_SUCCESS.getCode()
+					: description.getCode();
+			String moduleName = ModuleName.QUALITY_CLASSIFIER.toString();
+			registrationStatusService.updateRegistrationStatus(registrationStatusDto, moduleId, moduleName);
+			String eventId = isTransactionSuccessful ? EventId.RPR_402.toString() : EventId.RPR_405.toString();
+			String eventName = isTransactionSuccessful ? EventName.UPDATE.toString() : EventName.EXCEPTION.toString();
+			String eventType = isTransactionSuccessful ? EventType.BUSINESS.toString() : EventType.SYSTEM.toString();
+
+			auditLogRequestBuilder.createAuditRequestBuilder(description.getMessage(), eventId, eventName, eventType,
+					moduleId, moduleName, regId);
+
+		}
+
+		return object;
+	}
+
+
+	private iBioProviderApi getBioSdkInstance(BiometricType biometricType) throws BiometricException {
+		iBioProviderApi bioProvider = bioApiFactory.getBioProvider(biometricType, BiometricFunction.QUALITY_CHECK);
+		return bioProvider;
+	}
+	
+	private Map<String, String> getQualityTags(List<BIR> birs) throws BiometricException{
+		
+		Map<String, String> tags = new HashMap<String, String>();
+
+		// setting biometricNotAvailableTagValue for each modality in case biometrics are not available
+		if (birs == null) {
+			modalities.forEach(modality -> {
+				tags.put(qualityTagPrefix.concat(modality), biometricNotAvailableTagValue);
+			});
+			return tags;
+		}
+
+		HashMap<String, Float> bioTypeMinScoreMap = new HashMap<String, Float>();
+
+		// get individual biometrics file name from id.json
+		for (BIR bir : birs) {
+
+			if (bir.getOthers() != null) {
+				HashMap<String, String> othersInfo = bir.getOthers();
+				boolean exceptionValue = false;
+				if(othersInfo!=null) {

```

Comment:
why are we doing this change ? Is this associated to any JIRA bug id ?

Reply:
Jana added some changes in bio-utils,because of that it was not working in registration processor.Then we did some changes there I have added null check and some code changes.I reverted those,Null check i have not removed.Jana reverted his changes.